app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Api::V1::UsersController < ApplicationController
+  before_action :set_user, only: %i[show update destroy]
+
+  # GET /api/v1/users
+  def index
+    @users = User.all
+
+    render json: @users
+  end
+
+  # GET /api/v1/users/1
+  def show
+    render json: @user
+  end
+
+  # POST /api/v1/users
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      render json: @user, status: :created, location: api_v1_user_url(@user)
+    else
+      render json: @user.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /api/v1/users/1
+  def update
+    if @user.update(user_params)
+      render json: @user
+    else
+      render json: @user.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /api/v1/users/1
+  def destroy
+    @user.destroy
+    render json: { message: 'User deleted successfully.' }
+  end
```

Comment:
- [ ] What happens if this action fails?  Kindly, use some error handling in this action. 👍

Reply:
OK, I will add actions to show fails